dev/bazel/toolchains/cc_toolchain_lnx.bzl

```diff
@@ -154,6 +155,12 @@ def _add_gcc_toolchain_if_needed(repo_ctx, cc):
     else:
         return []
 
+def _add_sycl_linkage(repo_ctx, cc):
+    if ("icx" in cc) or ("icpx" in cc):
```

Comment:
i think we should check also for mode. i.e. we can use icx for CPU only mode. with this change we always would be adding fsycl

Reply:
Thanks for pointing this out. I pushed a commit which I think should fix it, by checking that the tooling has 'dpc' as done elsewhere in the script.

Although I'm not sure if that's the correct solution - it seems it will just assume everything needs DPCPP linkage whenever the compiler is 'icpx':
https://github.com/oneapi-src/oneDAL/blob/9b315cf8124c700f79bdc3007aac103df3b831bf/dev/bazel/toolchains/cc_toolchain.bzl#L23

which is also used for deciding flags in other parts of the build script:
https://github.com/oneapi-src/oneDAL/blob/9b315cf8124c700f79bdc3007aac103df3b831bf/dev/bazel/toolchains/cc_toolchain_lnx.bzl#L334

Reply:
I _suspect_ that the failing CI job after this last change:
```
.ci/scripts/test.sh: line 248: 46771 Aborted                 (core dumped) ${run_command} > "${e}".res 2>&1
terminate called after throwing an instance of 'daal::services::interface1::Exception'
  what():  NoErrorMessageFound
07:06:15 FAILED		pca_svd_dense_batch with errno 134
```

might have uncovered a different issue: it looks like the job `LinuxMakeGNU_OpenBLAS_x86` is executing SYCL examples, as those are auto-generated under `__release_lnx_gnu`.

The GCC compiler doesn't support SYCL so the examples shouldn't include them in that build, but apparently the job was somehow running them.

Reply:
examples/daal/cpp doesn't contain any SYCL/GPU related stuff.  I can't tell tis ether from log message. 

O would try rerunning public job again to start with. But it would be interesting to see results from internal runs as well